src/runtime_src/core/pcie/linux/shim.cpp

```diff
@@ -1515,14 +1515,15 @@ int xclLogMsg(xclDeviceHandle handle, xrtLogMsgLevel level, const char* tag, con
 {
     va_list args;
     va_start(args, format);
-    xocl::shim *drv = xocl::shim::handleCheck(handle);
-    int ret = drv ? drv->xclLogMsg(level, tag, format, args) : -ENODEV;
+    //Do NOT add device handle check
+    //This is logging function independent of device
+    //XMA passes NULL device handle to this API
```

Comment:
I am using xclLog() for internal logging in shim.cpp without the handle. Didn't realize I could pass NULL handle. But maybe we should make a new API as Max suggested? If not then I can remove xclLog and use xclLogMsg with NULL handle

Reply:
Let me check on aws change and the handle arg. Now this API is used in all emulation code as well. so need to check what all will be impacted.

Reply:
If it's only used in our code (including emulation), we can change them all. it should not be hard to do since we are removing arg, not adding one, which might be difficult. however, this is not the reason we can't change it. i believe that this api has been shipped with 2019.1, so it may be used by end users already. i think that if we want to change it, we'll need to mark existing one as obsolete and remove it in later release.

Reply:
ok. I will make that change.